Library/Homebrew/utils/curl.rb

```diff
@@ -26,8 +26,13 @@ def curl_args(options = {})
     args << "--silent" if !$stdout.tty? || ENV["TRAVIS"]
   end
 
-  args += options[:extra_args] if options[:extra_args]
-  args
+  extra_args = [*options[:extra_args]].map do |arg|
+    next arg.to_s if arg.is_a?(URI)
```

Comment:
Slightly uneasy about the explicit type check here — shouldn't all arguments be converted to strings for a shell command?